routes/root.js

```diff
@@ -12,136 +13,216 @@ const {
+    const [cubes, blogs] = await Promise.all([cubesq, blogsq]);
+
+    const cubeIds = [];
+    cubes.forEach(function(cube, index) {
+      cubeIds.push(cube._id);
+    });
+
+    const decks = await Deck.find({
+      cube: {
+        $in: cubeIds
+      }
+    }).sort({
+      'date': -1
+    }).limit(10);
+
+    return res.render('dashboard', {
+      posts: blogs,
+      cubes: cubes,
+      decks: decks,
+      loginCallback: '/'
+    });
+  } catch (err) {
+    console.log(err);
+    return res.status(500).send(err);
+  }
+});
 
-  Blog.find({
-    dev: 'true'
-  }).sort({
-    'date': -1
-  }).exec(function(err, result) {
-    if (err) {
-      blog = [];
-      console.log('blog failed to load');
+router.get('/dashboard/decks/:page', async function(req, res) {
+  try {
+    const pagesize = 30;
+    const page = req.params.page;
+    const user = await User.findById(req.user._id);
+    if (!user) {
+      return res.redirect('/landing');
     }
 
-    if (result) {
-      blog = result;
-    }
+    const cubes = await Cube.find({
+      owner: user._id
+    }).sort({
+      'date_updated': -1
+    }).exec();
 
-    routeReady();
-  });
+    const cubeIds = [];
+    cubes.forEach(function(cube, index) {
+      cubeIds.push(cube._id);
+    });
 
-  Deck.find().sort({
-    'date': -1
-  }).limit(10).exec(function(err, result) {
-    if (err) {
-      decks = [];
-      console.log('decks failed to load');
-    }
+    const decks = await Deck.find({
+      cube: {
+        $in: cubeIds
+      }
+    }).sort({
+      'date': -1
+    }).skip(pagesize * page).limit(30).exec();
+    const numDecks = await Deck.countDocuments({
+      cube: {
+        $in: cubeIds
+      }
+    }).exec();
 
-    if (result) {
-      decks = result;
+    var pages = [];
+    for (i = 0; i < numDecks / pagesize; i++) {
+      if (page == i) {
+        pages.push({
+          url: '/dashboard/decks/' + i,
+          content: (i + 1),
+          active: true
+        });
+      } else {
+        pages.push({
+          url: '/dashboard/decks/' + i,
+          content: (i + 1),
+        });
+      }
     }
 
-    routeReady();
-  });
+    return res.render('dashboard_decks', {
+      decks: decks,
+      pages: pages,
+      loginCallback: '/'
+    });
+  } catch (err) {
+    console.log(err);
+    return res.status(500).send(err);
+  }
 });
 
+router.get('/landing', async function(req, res) {
 
-//format: {search};{search};{search}:{page}
-//list like:
-//{property}{symbol}{value};
-//properties:
-//name, owner
-//symbols:
-//=,~(contains)
-router.get('/advanced_search', function(req, res) {
-  res.render('search/advanced_search', {
-    loginCallback: '/advanced_search'
+  const cubeq = Cube.countDocuments().exec();
+  const deckq = Deck.countDocuments().exec();
+  const userq = User.countDocuments().exec();
+
+  const [cube, deck, user] = await Promise.all([cubeq, deckq, userq]);
+
+  //this regex add commas to the number
+  res.render('landing', {
```

Comment:
DRY Nitpick: A bit more clear, not sure of the right naming, but something like:
```
//this regex add commas to the number
let threeNumbersFollowedByANumber = /\B(?=(\d{3})+(?!\d))/g;
...
numusers: user.toString().replace(threeNumbersFollowedByANumber, ",")
```

Reply:
Or better yet, user.toLocaleString('en-US')